src/birthder.js

```diff
@@ -274,6 +274,98 @@
         }
     }
 
+   /**
+    *Compare subarrays by month, day and then merge them
+    *
+    *@param {array} left - Subarray
+    *@param {array} right - Subarray
+    *@returns {array} - Sorted array
+    */
+    function merge(left, right) {
+        var result  = [];
+        var indexLeft = 0;
+        var indexRight = 0;
+
+        while (indexLeft < left.length && indexRight < right.length){
+            if (parseInt(left[indexLeft][0][1]) < parseInt(right[indexRight][0][1])){
+                result.push(left[indexLeft++]);
+            } else if (parseInt(left[indexLeft][0][1]) > parseInt(right[indexRight][0][1])) {
+                result.push(right[indexRight++]);
+            } else if (parseInt(left[indexLeft][0][0]) < parseInt(right[indexRight][0][0])) {
+                result.push(left[indexLeft++]);
+            } else {
+                result.push(right[indexRight++]);
+            }
+        }
+
+        return result.concat(left.slice(indexLeft)).concat(right.slice(indexRight));
+    };
+
+    /**
+    *Split array to subarrays and handle merge sort
+    *
+    *@param {array} items - Array of arrays [[dayOfBirthday, monthOfBirthday], username]
+    *returns {array} - Sorted array
+    */
+    function mergeSort(items) {
+        if (items.length < 2) {
+            return items;
+        }
+
+        const middle = Math.floor(items.length / 2);
+        const left = items.slice(0, middle);
+        const right = items.slice(middle);
+
+        return merge(mergeSort(left), mergeSort(right));
+    };
+
+    /**
+    *Implement binary search in sorted array
+    *
+    *@param {array} sortedUserArray - Sorted by month, day array of arrays or part of if it
+    *@returns {array} - Element of sortedUserArray with date before today
+    */
+    function search(sortedUserArray, monthToday, dayToday) {
+        if (sortedUserArray.length == 1) {
+            return sortedUserArray[0]
+        }
+
+        const middle = Math.floor(sortedUserArray.length / 2);
+        const month = parseInt(sortedUserArray[middle][0][1]);
+        const day = parseInt(sortedUserArray[middle][0][0]);
+
+        if (month > monthToday) {
+            return search(sortedUserArray.slice(0, middle), monthToday, dayToday)
+        } else if (month == monthToday) {
+             if (day >= dayToday) {
+                return search(sortedUserArray.slice(0, middle), monthToday, dayToday)
+            } else {
+                return search(sortedUserArray.slice(middle), monthToday, dayToday)
+            };
+        } else {
+            return search(sortedUserArray.slice(middle), monthToday, dayToday)
+        };
+    }
+
+    /**
+    *Reconstruct sortedUserArray in order of date today
+    *
+    *@param {array} sortedUserArray - Sorted by month, day array of arrays [[dayOfBirthday, monthOfBirthday], username]
```

Comment:
Use `{username: [dayOfBirthday, monthOfBirthday]}` instead of `[[dayOfBirthday, monthOfBirthday], username]`.